File: benchmark/output/python-pygame/openai/shooting.py
import pygame
import sys

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Vibe Game")
clock = pygame.time.Clock()

# Player variables
player_x, player_y = 384.0, 550.0
player_speed = 200

# Bullet variables
bullet_speed = 300
bullet_xs = []
bullet_ys = []

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        if event.type == pygame.KEYDOWN:
            if event.key == pygame.K_SPACE:
                bullet_xs.append(player_x + 16)  # Center of the player
                bullet_ys.append(player_y)

    keys = pygame.key.get_pressed()
    if keys[pygame.K_LEFT]:
        player_x -= player_speed * dt
    if keys[pygame.K_RIGHT]:
        player_x += player_speed * dt

    # Update bullet positions
    for i in range(len(bullet_ys)):
        bullet_ys[i] -= bullet_speed * dt

    screen.fill((0, 0, 0))
    # Draw player
    pygame.draw.rect(screen, (255, 255, 255), (int(player_x), int(player_y), 32, 32))
    # Draw bullets
    for i in range(len(bullet_xs)):
        pygame.draw.rect(screen, (255, 255, 255), (int(bullet_xs[i]), int(bullet_ys[i]), 4, 4))

    pygame.display.flip()

pygame.quit()
sys.exit()

File: benchmark/output/lua-love/openai/shooting.lua
local player_x = 384
local player_y = 550
local player_speed = 200

local bullet_xs = {}
local bullet_ys = {}
local bullet_speed = 300

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Vibe Game")
end

function love.update(dt)
    if love.keyboard.isDown("left") then
        player_x = player_x - player_speed * dt
    end
    if love.keyboard.isDown("right") then
        player_x = player_x + player_speed * dt
    end

    for i = 1, #bullet_ys do
        bullet_ys[i] = bullet_ys[i] - bullet_speed * dt
    end
end

function love.keypressed(key)
    if key == "space" then
        table.insert(bullet_xs, player_x + 16)
        table.insert(bullet_ys, player_y)
    end
end

function love.draw()
    love.graphics.rectangle("fill", player_x, player_y, 32, 32)
    
    for i = 1, #bullet_xs do
        love.graphics.rectangle("fill", bullet_xs[i], bullet_ys[i], 4, 4)
    end
end